low-level/files/opt/seagate/sspl/bin/sspl_config

```diff
@@ -185,15 +185,35 @@ case $cmd in
         ;;
 esac
 
+# Get the version. Output can be 3.3.5 or 3.8.9 or in this format
+rabbitmq_version=$(rpm -qi rabbitmq-server | awk -F': ' '/Version/ {print $2}')
+
+# Get the Major release version parsed. (Eg: 3 from 3.8.9)
+rabbitmq_major_release=$(echo "${rabbitmq_version:0:-3}" | sed 's/.$//')
+
+# Get the Minor release version parsed. (Eg: 3.8 from 3.8.9)
+rabbitmq_minor_release=$(echo "${rabbitmq_version::-1}" | sed 's/.$//')
+
+# Get the Maitenance release version parsed from minor release. (Eg: 8 from 3.8)
+rabbitmq_maintenance_release=$(echo "${rabbitmq_minor_release:2:3}")
+
 # Skip this step if sspl is being configured for node replacement scenario as consul data is already
 # available on healthy node
 # Updating RabbitMQ cluster nodes.
 # In node replacement scenario, avoiding feeding again to avoid over writing already configured values
 # with which rabbitmq cluster may have been created
 [ -f $REPLACEMENT_NODE_ENV_VAR_FILE ] || {
     # Update cluster_nodes key in consul
-    out=$(rabbitmqctl cluster_status | grep running_nodes | cut -d '[' -f2 | cut -d ']' -f1 | sed 's/rabbit@//g' | sed 's/,/, /g')
-    pout=$(echo $out | sed  "s/'//g" | sed  "s/ //g")
+    if [ $rabbitmq_major_release -eq 3 ] && [ $rabbitmq_maintenance_release -eq 8 ];
+    then
+        rabbitmq_cluster_status=$(/usr/sbin/rabbitmqctl cluster_status --formatter json)
+        cluster_running_nodes=$(python3 -c "rabbitmq_status=$rabbitmq_cluster_status; print(rabbitmq_status['running_nodes'])")
+        pout=$(echo $cluster_running_nodes | sed 's/rabbit@//g' | cut -d '[' -f2 | cut -d ']' -f1 | tr -d \'\" | tr -d ' ')
+    else
```

Comment:
this may get applied to any version now, above 3.8.9 or below, and we intend to keep this for backward compatibility with 3.3.5 only, so better put that explicit check or fail out early with proper clear error string.

Reply:
Can we avoid this parsing based on the version? Instead we can get what we back in the cluster status output: we could get:
running_nodes OR
Running Nodes

Then we can parse accordingly?

Reply:
In the previous patch, that is why i kept it -ge (greater that or equal to). And they are giving this option --formatter json which must be with purpose (to get the output in json format which is widely accepted). So, I don't think that they will remove this option from next version.
With your suggested check, It will not work then if we go for other some check and eventually we will add more if else statements...

Reply:
@indrajitzagade Earlier version it was running_version, Now it is Running Nodes, tomorrow it can be else. 
From 3.8.9, they are giving this --formatter option in order to get the cluster status format in a united manner and json is widely used and easy for parsing. So, I think we should use that one.

Reply:
oh ok @Madhura-08 ,this --formatter option looks useful here.

Reply:
> From 3.8.9, they are giving this --formatter option in order to get the cluster status format in a united manner

--formatter with json choice, how it looks. And since they are changing with different versions, its not sure they would not change in future too, so having 'ge' may or may not work and we wont even know from logs what went wrong, which am trying to address here, by having version specific checks and beyond that give warning log, may be try once extracting running nodes and if failed, then come out OR come out directly without trying.

Also for 3.3.5 older version, this option is not there right, so we need to have that version based parsing still i feel.

Reply:
>Also for 3.3.5 older version, this option is not there right, so we need to have that version based parsing still i feel.
Right.
So, my check is:
if Version is >= 3.8.9
   use --formatter
else:
    use old method
So, 3.3.5 check will be satisfied here. 
And since they are changing with different versions
I think, its something additional they have added with some purpose. So, I don't think so, they will remove that thing in the next version.
And, suppose i added that as per your suggestion, it will be like below:
if Version is == 3.8.9
   use --formatter
else:if version == 3.3.5 (or may be <=):
    use old method
else:
    ??
What will be in else ??

Reply:
>if Version is == 3.8.9
use --formatter

Agree

>else:if version == 3.3.5 (**dont add <= where we dont know the behavior really**):
use old method
else:
??
What will be in else ??

failout with an error log string

Reply:
Ok. I will do as you said then.